src/serverenvironment.cpp

```diff
@@ -1066,6 +1066,87 @@ bool ServerEnvironment::swapNode(v3s16 p, const MapNode &n)
 	return true;
 }
 
+u8 ServerEnvironment::findSunlight(v3s16 pos) const
+{
+	// Returns a unique 64 bit unsigned int for a v3s16 position
+	auto get_node_position_key = [&](v3s16 pos) -> u64 {
+		return pos.Z * 0x10000uLL * 0x10000uLL
```

Comment:
pos.Z can be negative, which I believe would result in an overflow to u64 (assuming two's compliment)

The Lua version of this offsets by 1/2 the range, to ensure 0-65536:

```lua
function core.hash_node_position(pos)
	return (pos.z + 32768) * 65536 * 65536
		 + (pos.y + 32768) * 65536
		 +  pos.x + 32768
end
```

Reply:
`MapDatabase::getBlockAsInteger` does this calculation as well, but `s16` should be casted to `u64` first, for additional safety.

Reply:
It uses `getBlockAsInteger` now. The function takes a `s16` vector.